db/fixtures/ae_datastore/ManageIQ/Deployment/ContainerProvider/System/StateMachine/Deployment.class/__methods__/pre_deployment.rb

```diff
@@ -1,93 +1,70 @@
+  commands
 end
 
 def pre_deployment
-  commands = ['sudo yum install -y ansible',
+  commands = ['sudo yum install -y ansible-1.9.4',
               'sudo yum install -y openshift-ansible openshift-ansible-playbooks pyOpenSSL',
-              "sudo mv ~/inventory.yaml /usr/share/ansible/openshift-ansible/"
-              ]
+              "sudo mv ~/inventory.yaml /usr/share/ansible/openshift-ansible/",
+              "sudo yum install -y atomic-openshift-utils"
+  ]
   $evm.log(:info, "********************** #{$evm.root['ae_state']} ***************************")
-  begin
-    Net::SSH.start($evm.root['deployment_master'], $evm.root['user'], :paranoid => false, :forward_agent => true,
-                   :key_data => $evm.root['private_key']) do |ssh|
-      $evm.log(:info, "Connected to deployment master, ip address: #{$evm.root['deployment_master']}")
-      system "scp -o 'StrictHostKeyChecking no' inventory.yaml #{$evm.root['user']}@#{$evm.root['deployment_master']}:~/"
-      failed_execute = false
-      release = ssh.exec!("cat /etc/redhat-release")
-      if release.include?("CentOS")
-        commands.unshift("sudo yum install epel-release -y",
-                         "sudo curl -o /etc/yum.repos.d/maxamillion-origin-next-epel-7.repo #{REPO_URL}")
-      elsif release.include?("Red Hat Enterprise Linux") &&
-            !$evm.root['automation_task'].automation_request.options[:attrs][:containerized]
-        commands = handle_rhel_subscriptions(commands)
-      end
-      commands.each do |cmd|
-        result = ssh_exec!(ssh, cmd)
-        unless result[:exit_code] == 0
-          $evm.root['automation_task'].message = "FAILED: couldn't execute command #{cmd}. ERROR: #{result[:stderr]}"
-          $evm.root['ae_result'] = "error"
-          failed_execute = true
-        end
-        break if failed_execute
-      end
-      unless failed_execute
-        $evm.root['ae_result'] = "ok"
-        $evm.root['automation_task'].message = "successful pre-deployment"
-      end
-    end
-  rescue
-    $evm.root['ae_result'] = "error"
-    $evm.root['automation_task'].message = "Cannot connect to deployment master " \
-                                           "(#{$evm.root['deployment_master']}) via ssh"
-  ensure
-    $evm.log(:info, "State: #{$evm.root['ae_state']} | Result: #{$evm.root['ae_result']} "\
-             "| Message: #{$evm.root['automation_task'].message}")
+  $evm.root['container_deployment'].perform_scp($evm.root['deployment_master'], $evm.get_state_var(:ssh_user), "inventory.yaml", "inventory.yaml")
+  release = $evm.root['container_deployment'].perform_agent_commands($evm.root['deployment_master'], $evm.get_state_var(:ssh_user), ["sudo cat /etc/redhat-release"])[:stdout]
+  if release.include?("CentOS")
+    commands.unshift("sudo yum install epel-release -y",
```

Comment:
Seems `commands` will get executed afer this if.
Can you run them on the spot instead of stashing for later?